wpilibj/src/athena/java/edu/wpi/first/wpilibj/RobotDrive.java

```diff
@@ -580,14 +565,15 @@ public void setLeftRightMotorOutputs(double leftOutput, double rightOutput) {
   /**
    * Limit motor values to the -1.0 to +1.0 range.
    */
-  protected static double limit(double num) {
-    if (num > 1.0) {
-      return 1.0;
-    }
-    if (num < -1.0) {
-      return -1.0;
-    }
-    return num;
+  protected static double limit(double number) {
+    return Math.max(Math.min(number, 1.0), -1.0);
+  }
+
+  /**
+   * Squares the number but preserves the sign.
+   */
+  protected static double squarePreserveSign(double number) {
+    return Math.copySign(Math.pow(number, 2), number);
```

Comment:
I would think `number * number` would be slightly faster than `pow(number, 2)`, at least in Java (C++ template optimization means C++ likely would not see a performance hit).